default/python/universe_generation/empires.py

```diff
@@ -113,18 +113,24 @@ def find_home_systems_for_min_jump_distance(self, systems_pool, min_jumps):
             if system not in self.system_merit:
                 self.system_merit[system] = calculate_home_system_merit(system)
 
-        best_case_worst_merit = sorted(self.system_merit.values())[self.num_home_systems - 1]
-        current_worst_merit = 0
+        # In the list of systems sorted by merit, if the first num_home_systems are all min_jumps
+        # apart, then the merit of the num_home_systems nth system is the
+        # best_case_merit_lower_bound that can be achieved.
+        best_case_merit_lower_bound = sorted(self.system_merit.values())[self.num_home_systems - 1]
```

Comment:
> I'm not saying that there can't be a better name, but best_case_merit_upper_bound IMO doesn't tell me that it's the lowest value from a set...?

The comments and / or docstrings need to explain what are the criteria used to calculate the "merit" of the selected systems once in the very beginning when explaining the algorithm. 
In the PR it is stated as "Use the minimum merit of the best num_home_system systems found  as the merit of the current best set of systems."

So what is the merit of a selection of systems is already clearly stated. There is absolutely no need to state it anywhere again especially not in a variable name. If you want to avoid confusion between single-system-merit and set-of-systems-merit, then do not use "merit" twice but switch to "score"/"rating" or whatever else comes to mind so that it reads
"The score of a selection of systems is defined as the lowest merit of the systems within that selection."

If you then call it ```score_upper_bound```, there will be no ambiguity. If you want to explain how the upper bound is calculated, then that should be a single line comment at the location where it is calculated but not a ```4_underscore_variable_name_to_state_already_stated_facts_that_are_never_needed_again``` is unneccessary and very clumsy IMO.